src/main/kotlin/app/revanced/patches/youtube/misc/video/quality/patch/DefaultVideoQualityPatch.kt

```diff
@@ -0,0 +1,140 @@
+import app.revanced.patches.youtube.misc.video.videoid.patch.VideoIdPatch
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.reference.FieldReference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, VideoIdPatch::class, SettingsPatch::class])
+@Name("default-video-quality")
+@Description("Adds the ability to remember the video quality.")
+@DefaultVideoQualityCompatibility
+@Version("0.0.1")
+class DefaultVideoQualityPatch : BytecodePatch(
+    listOf(
+        VideoQualitySetterFingerprint
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        val (keys, values) = buildMap {
+            fun add(quality: String, entry: String, entryValue: String) {
+                fun String.withValue(value: String) = StringResource(this, value)
+
+                put(
+                    "revanced_video_quality_${quality}_entry".withValue(entry),
+                    "revanced_video_quality_${quality}_entryValue".withValue(entryValue)
+                )
+            }
+
+            add("auto", "Auto", "-2")
+
+            arrayOf("144", "240", "360", "480", "720", "1080", "1440", "2160").forEach {
+                add(it, "${it}p", it)
+            }
+        }.let { map ->
+            map.keys.toList() to map.values.toList()
+        }
+
+        SettingsPatch.PreferenceScreen.MISC.addPreferences(
+            PreferenceScreen(
+                "revanced_default_video_quality",
+                StringResource("revanced_default_video_quality_title", "Video quality settings"),
+                listOf(
+                    SwitchPreference(
+                        "revanced_remember_video_quality_selection",
+                        StringResource(
+                            "revanced_remember_video_quality_selection_title",
+                            "Remember current video quality"
+                        ),
+                        true,
+                        StringResource(
+                            "revanced_remember_video_quality_selection_summary_on",
+                            "The current video quality will not change"
+                        ),
+                        StringResource(
+                            "revanced_remember_video_quality_selection_summary_off",
+                            "Video quality will be remembered until a new quality is chosen"
+                        )
+                    ),
+                    ListPreference(
+                        "revanced_default_video_quality_wifi",
+                        StringResource(
+                            "revanced_default_video_quality_wifi_title",
+                            "Change video quality on Wi-Fi network"
+                        ),
+                        ArrayResource("revanced_video_quality_wifi_entries", keys),
+                        ArrayResource("revanced_video_quality_wifi_entryValues", values),
```

Comment:
This key should be named in snake casing:

```suggestion
                        ArrayResource("revanced_video_quality_wifi_entry_values", values),
```

Make sure to apply the change throughout the integrations as well.